pkg/identity/numericidentity.go

```diff
@@ -168,6 +168,11 @@ func (id NumericIdentity) IsReservedIdentity() bool {
 	return isReservedIdentity
 }
 
+// ClusterID returns the cluster ID associated with the identity
+func (id NumericIdentity) ClusterID() int {
```

Comment:
Add a unit test for this

Reply:
Tracked: https://github.com/cilium/cilium/issues/4999